### ISSUE TITLE
testnet.programmingbitcoin.com isn't responding anymore

### ISSUE THREAD

**Issue by tindandelion** [jimmysong/programmingbitcoin#245]
As of now, May 26th, I can't reach `testnet.programmingbitcoin.com` on port 18333 anymore. The server isn't responding.

**Comment by Mynima** [comment]
Not sure if this is related but I'm getting a timeout error on the testnet (Chapter10, exercise 5):

TimeoutError: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond

Same is actually happening on the mainnet (mainnet.programmingbitcoin.com) for the very last piece of example code in Chapter10.

**Comment by maestrovt** [comment]
I have the same issue as of 8/7/2022. If I run the first example in Chapter 7, the code fails due to testnet.programmingbitcoin.com returning "502 Bad Gateway".

```
ValueError: unexpected response: <html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx/1.21.6</center>
</body>
</html>
```

**Comment by fedetaladriz** [comment]
I have the same problem. @tindandelion  @maestrovt Did you guys manage to solve it?

**Comment by maestrovt** [comment]
@fedetaladriz No, not yet. I have reached out to Jimmy, no response as yet, I will post here if I get an answer.

**Comment by tindandelion** [comment]
For my experiments with Bitcoin protocol, I ended up using another gateway: `testnet-seed.bluematt.me`, port 18333

**Comment by fedetaladriz** [comment]
I managed to solve it for the mainnet (`mainnet.programmingbitcoin.com`) following [this](https://github.com/zeltsi/byob/blob/master/p2p/version.py) example. The testnet seems to be not working.

**Comment by nicolafiorillo** [comment]
For those who still need it: `https://blockchain.info/rawtx/{tx_ID}?cors=true&format=hex`

**Comment by SDZZGNDRC** [comment]
> I managed to solve it for the mainnet (`mainnet.programmingbitcoin.com`) following [this](https://github.com/zeltsi/byob/blob/master/p2p/version.py) example. The testnet seems to be not working.

The mainnet `mainnet.programmingbitcoin.com` seems to be not working now too.

**Comment by nicolafiorillo** [comment]
> > I managed to solve it for the mainnet (`mainnet.programmingbitcoin.com`) following [this](https://github.com/zeltsi/byob/blob/master/p2p/version.py) example. The testnet seems to be not working.
> 
> The mainnet `mainnet.programmingbitcoin.com` seems to be not working now too.

Use blockstream.info as alternative:
https://github.com/Blockstream/esplora/blob/master/API.md
https://github.com/Blockstream/esplora/blob/master/API.md#post-tx

**Comment by edyu** [comment]
I can verify that testnet.programming.com doesn't work anymore. The mainnet still does but not testnet. For transactions there are alternative site but for the handshake and bloomfilters, I can't find another alternative other than setting up my own testnet.

**Comment by edyu** [comment]
Looking at dns, I believe mainnet.programmingbitcoin.com and testnet.programmingbitcoinc.om point to the same server/proxy so the only difference is the port. However, my code works with mainnet but not with testnet. Not sure why.
